onnx/defs/math/defs.cc

```diff
@@ -2178,17 +2178,18 @@ void einsumRankInference(
 
   // Parse the left-hand side
   std::stringstream str(left_equation);
-  while (std::getline(str, term, ',')) {
+  while (!str.eof()) {
+    std::getline(str, term, ',');
     auto ellipsis_index = term.find("...");
+    if (numInputs <= num_operands) {
+      fail_shape_inference(
+          "Number of input tensors does not match the operands in the equation.");
+    }
+    size_t rank =
+    ctx.getInputType(num_operands)->tensor_type().shape().dim_size();
```

Comment:
Is the formatting produced by clang? Looks odd ... I expected some indentation of the second line.

Reply:
No, I just updated it